lightning-invoice/src/utils.rs

```diff
@@ -50,14 +50,97 @@ use sync::Mutex;
 /// [`ChannelManager::get_phantom_route_hints`]: lightning::ln::channelmanager::ChannelManager::get_phantom_route_hints
 /// [`PhantomRouteHints::channels`]: lightning::ln::channelmanager::PhantomRouteHints::channels
 pub fn create_phantom_invoice<Signer: Sign, K: Deref>(
-	amt_msat: Option<u64>, description: String, payment_hash: PaymentHash, payment_secret:
-	PaymentSecret, phantom_route_hints: Vec<PhantomRouteHints>, keys_manager: K, network: Currency
-) -> Result<Invoice, SignOrCreationError<()>> where K::Target: KeysInterface {
+	amt_msat: Option<u64>,
```

Comment:
I'm also really not a huge fan of making this layout super vertical just to call a utility method.